x-pack/plugins/synthetics/common/runtime_types/monitor_management/locations.ts

```diff
@@ -53,14 +53,19 @@ export const ManifestLocationCodec = t.interface({
   status: LocationStatusCodec,
 });
 
-export const ServiceLocationCodec = t.interface({
-  id: t.string,
-  label: t.string,
-  geo: LocationGeoCodec,
-  url: t.string,
-  isServiceManaged: t.boolean,
-  status: LocationStatusCodec,
-});
+export const ServiceLocationCodec = t.intersection([
+  t.interface({
+    id: t.string,
+    label: t.string,
+    geo: LocationGeoCodec,
+    url: t.string,
+    isServiceManaged: t.boolean,
+    status: LocationStatusCodec,
+  }),
+  t.partial({
```

Comment:
Since these are public locations, can public locations have a notion of being invalid? I see that we also have a private locations type.

Reply:
i think there is no harm in it. Otherwise we  will have to declare a new type. Which seems too disruptive at this point.